Extra features for popup gui

SimonFlapse opened the pull request:
Added 13 new icons to the popup gui
Added the ability to change title

Updated popup-update commands wording, title and icon.

Closes #636

SimonFlapse committed:
Extra features for popup gui

Added 13 new icons to the popup gui
Added the ability to change title

Updated popup-update commands wording, title and icon.

SimonFlapse committed:
Porting to fix merge conflict

SimonFlapse committed:
Merge branch 'develop' into Popup-Extended

linaori commented on features/gui/popup.lua:
```diff
@@ -5,7 +5,28 @@ local Command = require 'utils.command'
 
-local function show_popup(player, message)
+local icons = {
+    "ammo_icon",                            -- 1
```
Would be nice if we could refer to them externally:
```lua
show_popup(player, 'foo', 'bar', Popup.icons.danger_icon)
```

SimonFlapse replied:
I've added the ability to provide a sprite path, while keeping the "shortcuts" for the 14 similar icons

grilledham commented on features/gui/popup.lua:
```diff
@@ -5,7 +5,28 @@ local Command = require 'utils.command'
+    "too_far_from_roboport_icon"            -- 14
+}
+
+local function show_popup(player, message, title, icon_id)
```
What if we change `icon_id` to `sprite_path` then use linaori's sugestion above. This would allow use to use any sprite not just the ones provided by the module

SimonFlapse committed:
Allowing sprite paths instead of icon_id

Added a pass-through for sprite paths while keeping the 14 "quick" access icons.

plague006 commented:
Is https://github.com/Refactorio/RedMew/pull/687#pullrequestreview-196601458 resolved?

And when you get a chance can you clean up the lint warnings?

SimonFlapse committed:
fixing lint warnings

SimonFlapse committed:
Brainfarted, forgot to change a variable

SimonFlapse commented:
There you go

SimonFlapse committed:
Addon to prevent popup spam

plague006 commented:
Looks good to me, merging with my thanks.
